src/modules/core/components/MembersList/SortingRow/SortingRow.tsx

```diff
@@ -19,22 +28,57 @@ const MSG = defineMessages({
   },
 });
 
-const SortingRow = () => {
+const SortingRow = ({ handleSortingMethodChange, sortingMethod }: Props) => {
+  const nextSortingByRepMethod = useMemo(() => {
+    switch (sortingMethod) {
+      case SORTING_METHODS.BY_HIGHEST_REP:
+        return SORTING_METHODS.BY_LOWEST_REP;
+      default:
+        return SORTING_METHODS.BY_HIGHEST_REP;
+    }
+  }, [sortingMethod]);
+  const nextSortingByRoleMethod = useMemo(() => {
+    switch (sortingMethod) {
+      case SORTING_METHODS.BY_HIGHEST_ROLE_ID:
+        return SORTING_METHODS.BY_LOWEST_ROLE_ID;
+      case SORTING_METHODS.BY_LOWEST_ROLE_ID:
+        return SORTING_METHODS.BY_HIGHEST_REP;
+      default:
+        return SORTING_METHODS.BY_HIGHEST_ROLE_ID;
+    }
+  }, [sortingMethod]);
+  const isSortingByRep =
+    sortingMethod === SORTING_METHODS.BY_HIGHEST_REP ||
+    sortingMethod === SORTING_METHODS.BY_LOWEST_REP;
   return (
     <div className={styles.container}>
-      <Button className={styles.sortingButton}>
+      <Button
+        className={styles.sortingButton}
+        onClick={() => handleSortingMethodChange(nextSortingByRoleMethod)}
+      >
         <FormattedMessage {...MSG.permissions} />
         <Icon
-          className={styles.sortingIcon}
-          name="caret-down-small"
+          className={classnames(styles.sortingIcon, {
+            [styles.toggledIcon]: !isSortingByRep,
+          })}
+          name={`caret-${
+            sortingMethod === SORTING_METHODS.BY_HIGHEST_ROLE_ID ? 'up' : 'down'
+          }`}
           title={MSG.permissions}
         />
       </Button>
-      <Button className={styles.sortingButton}>
+      <Button
+        className={styles.sortingButton}
+        onClick={() => handleSortingMethodChange(nextSortingByRepMethod)}
+      >
         <FormattedMessage {...MSG.reputation} />
         <Icon
-          className={styles.sortingIcon}
-          name="caret-down-small"
+          className={classnames(styles.sortingIcon, {
+            [styles.toggledIcon]: isSortingByRep,
+          })}
+          name={`caret-${
```

Comment:
It's a nitpick but I think it would be easier to read without string literal, it's ok to repeat caret: 
`sortingMethod === SORTING_METHODS.BY_HIGHEST_REP ? 'caret-up' : 'caret-down'`